src/utils/channel_utils.js

```diff
@@ -22,7 +22,7 @@ export function buildDisplayableChannelList(usersState, allChannels, myPreferenc
     const missingDirectChannels = createMissingDirectChannels(usersState.currentUserId, allChannels, myPreferences);
 
     const {currentUserId, profiles} = usersState;
-    const locale = profiles && profiles[currentUserId] ? profiles[currentUserId].locale : 'en';
+    const locale = profiles && profiles[currentUserId] && profiles[currentUserId].locale ? profiles[currentUserId].locale : 'en';
```

Comment:
looks like this code is used in three different places in this PR, can you make this a function?